src/dataset.js

```diff
@@ -30,13 +30,16 @@ var Table = require('./table.js');
  * @class
  * @param {BigQuery} bigQuery {@link BigQuery} instance.
  * @param {string} id The ID of the Dataset.
+ * @param {object} [metadata] Dataset metadata.
```

Comment:
We should rename this to `options` and `options.location`. It shouldn't have a connection to "Dataset metadata" from the user's perspective, because that indicates it has a connection with the remote, upstream object. For example, the user could provide other Dataset metadata, and assume that it will be used in API responses and update that value. We just want the user to provide a `location`, if they need to, through a configuration object, like we usually do;

```js
* @param {object} [options] Configuration object.
* @param {string} [options.location] The geographic...
```

Reply:
Done.